### ISSUE TITLE
Feature request: Allow login with service principal (or enable 'additional options' to pass login, similar to Dax studio)

### ISSUE THREAD

**Issue by atmisra** [microsoft/Analysis-Services#139]

**Comment by christianwade** [comment]
Hi, thanks for the feedback. Are you referring to ALM Toolkit? The login/authorization for ALM Toolkit is done by the AS client libraries, so SPN support is based on the platform, not ALM Toolkit. If you can log in with SSMS, you will be able to log in with ALM Toolkit.

**Comment by atmisra** [comment]
Yes - ALM toolkit. The issue I encounter is that SPN is a user/app id and password, and i see no place to be able to pass this into ALM toolkit (only microsoft authentication pop-up, which doesn't allow this entry as far as I know). 

For example, in DAX studio, I can add the credentials to the additional options section:
![image](https://user-images.githubusercontent.com/80759879/156703781-c0aa82a6-b476-4baa-81bb-f31284521dd9.png)

Using the connection string:
https://docs.microsoft.com/en-us/analysis-services/instances/connection-string-properties-analysis-services?view=asallproducts-allversions#user-idpassword

**Comment by christianwade** [comment]
Please take a look at this article: https://docs.microsoft.com/power-bi/admin/service-premium-service-principal

The section section connection strings for XMLA endpoint shows the required format
https://docs.microsoft.com/power-bi/admin/service-premium-service-principal#connection-strings-for-the-xmla-endpoint

User ID=app:<appId>@<tenantId>
Password=<app_secret>

You should be able to test these username/password values for SSMS.

**Comment by atmisra** [comment]
Apologies if I am not being clear. I am familiar with how to build the connection string, and how to log in to SSMS using SPN. What I am unable to see, is any location where I can enter this connection string in ALM toolkit (when selecting source and target models).

**Comment by christianwade** [comment]
That's OK. You shouldn't need to create a connection string. It should work from SSMS using the id/password fields if you use the format above - without creating a connection string.

**Comment by christianwade** [comment]
Sorry for the delayed response. Can you connect from SSMS? If no, can you look into these docs?
https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-service-principal
https://docs.microsoft.com/en-us/power-bi/enterprise/service-premium-connect-tools#service-principals

There are a few extra steps required to set up SPNs for automation in Power BI compared to AAS.

**Comment by maturmel** [comment]
Hi,

SSMS when trying to connect like below:
![image](https://user-images.githubusercontent.com/13457371/176442866-4e04abb7-acdb-4875-a7af-e4e62f87fbd9.png)

Gives me the 401 Unauthorized. So it looks like there is a missing permission somewhere in the Azure AD setup, then?

What's weird is that calling the Get-PowerBIAccessToken cmdlet does not return any error, however.
I'll go through the docs again and try to find out what part I did wrong. 

Thanks!

**Comment by maturmel** [comment]
Hi @christianwade ,

I found out the root cause. I think the service account user is member of my security group, but not the service principal itself. And in Power BI Admin center, I had allowed API access limited to my security group only. When I enable access for entire org, the connection works in SSMS. So it ends up being a question of membership setup. I will work that through with our security admins.

Reviewing the documents made me realize this. Thanks again for your reply.
